File: leetcode/no1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var getDecimalValue = function (head) {
    let numArray = [];
    while (head != null) {
        numArray.push(head.val);
        head = head.next;
    }

    var digit = parseInt(numArray.join(""), 2);
    return digit;
};

File: kotlin/src/main/kotlin/no1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.kt
import ListNode.ListNode

fun main() {

    /**
     * Example:
     * var li = ListNode(5)
     * var v = li.`val`
     * Definition for singly-linked list.
     * class ListNode(var `val`: Int) {
     *     var next: ListNode? = null
     * }
     */
    class Solution {
        fun getDecimalValue(head: ListNode?): Int {

            var numString = ""
            var node = head

            while (node != null) {
                numString += node.`val`
                node = node.next
            }
            return Integer.parseInt(numString, 2)
        }
    }
}